tokio/src/net/udp/socket.rs

```diff
@@ -303,6 +336,38 @@ impl UdpSocket {
             .await
     }
 
+    /// Attempts to receive a single datagram message on the socket from the remote
+    /// address to which it is `connect`ed.
+    ///
+    /// The [`connect`] method will connect this socket to a remote address. The future
+    /// will resolve to an error if the socket is not connected..
+    ///
+    /// # Return value
+    ///
+    /// The function returns:
+    ///
+    /// * `Poll::Pending` if the socket is not ready to read
+    /// * `Poll::Ready(Ok(n))` `n` is the number of bytes read.
+    /// * `Poll::Ready(Err(e))` if an error is encountered.
+    ///
+    /// # Errors
+    ///
+    /// This function may encounter any standard I/O error except `WouldBlock`.
+    ///
+    /// [`connect`]: method@Self::connect
+    pub fn poll_recv(&self, cx: &mut Context<'_>, buf: &mut [u8]) -> Poll<io::Result<usize>> {
```

Comment:
Shouldn't `buf` be the new `ReadBuf` here?

Reply:
Good point. Same for `poll_recv_from`.

Reply:
Ah okay. For converting `ReadBuf` -> `&mut buf` there's a few methods that we could use. `initialize_unfilled` which is safe and initializes the buffer, and `unfilled_mut` which is unsafe and assumes it's initialized. Methods in `TcpStream` use the unsafe version but they are private, if that makes any difference. I've used the safe versions initially here.

Reply:
Well `ReadBuf` would allow receiving into uninitialized memory. I'm pretty sure the underlying IO primitives should support that. Even if you prefer to write a version that initializes it first, defining it with `ReadBuf` would be necessary for adding that optimization later without a breaking change.

Reply:
Alright, updated.